Tasks/words.js

```diff
@@ -1,41 +1,14 @@
 // Count words in a string
-
-Words = function (s) {
-  number_of_words_in_s = 0;
-  flag = false;
-  for (c of s) {
-    if (!flag) {
-      if (c === ' ') {
-        if (flag === true) {
-          flag = false;
-        } else {
-          flag = false;
-        }
-      } else {
-        if (flag === true) {
-          flag = true;
-        } else {
-          flag = true;
-        }
-        number_of_words_in_s++;
-      }
-    } else {
-      if (c === ' ') {
-        if (flag === true) {
-          flag = false;
-        } else {
-          flag = false;
-        }
-      } else {
-        if (flag === true) {
-          flag = true;
-        } else {
-          flag = true;
-        }
-      }
+//step6 change if, else if to ternar operator
+const words = (s) => {
+  let wordsCount = 0;
+  let wordFound = false;
+  for (let i = 0; i < s.length; i++) {
+    s[i] === ' ' ? inWord = false : !inWord ? wordsCount++ : null;
```

Comment:
Use ternary only to calculate expression to be assigned/passed somewhere not for branching logic